File: src/leetCode/java/Solution217.java
package leetCode.java;

import java.util.*;

//O(n) O(1)
class Solution217 {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> memo = new HashSet<>();
        for(int num : nums){
            if(memo.contains(num)){
                return true;
            }else{
                memo.add(num);
            }
        }
        return false;
    }
}

File: src/leetCode/c++/Solution217.cpp
//O(n) O(n)
class Solution217 {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> memo;
        for(int num : nums){
            if(memo.count(num)){
                return true;
            }else{
                memo.insert(num);
            }
        }
        return false;
    }
};